src/convertToObject.js

```diff
@@ -6,7 +6,32 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const mainObj = {};
+
+  const workArr = sourceString.split(';');
+
+  const cleanArr = workArr
+    .map(function (item) {
+      const result = item.replace(/\n/g, '').trim();
+
+      return result;
+    })
+    .filter((item) => item !== '');
```

Comment:
Instead of using a `map` followed by a `filter`, you can combine these operations into a single `reduce` to streamline the code and avoid iterating over the array twice.